mantle/cmd/kola/kola.go

```diff
@@ -600,7 +602,9 @@ func getParentFcosBuildBase() (string, error) {
 		return "", errors.New("no ref in build metadata")
 	}
 
-	stream := filepath.Base(kola.CosaBuild.Meta.BuildRef)
+	if stream == "" {
+		stream = filepath.Base(kola.CosaBuild.Meta.BuildRef)
```

Comment:
Nit: it would be better not shadow-assigning an input argument.

Reply:
Hm, why?

Reply:
I think I'd agree with that guideline in general for complex functions with many disparate arguments. Though not sure if it's worth it here. IMO, it's less mental load to keep the same number of variables when reading through than introducing new but slightly different ones.

Hmm, though maybe it's cleaner anyway here to have the caller of `getParentFcosBuildBase` always pass the stream. OK, done!

Reply:
Perhaps personal taste, but it diminishes the path of least-surprise for no real reason.

Not specifically for this code, but in general: our future selves won't be positively surprised when debugging this, tracing the input argument, following a function call at the bottom of the body, and then wondering why there seem to be mismatch/mutation in the input argument.

Reply:
It's a fairly common pattern in Rust at least to do this "rebinding" at least for local variables IME.